gen_planet.py

```diff
@@ -45,50 +140,121 @@ def gen_plane():
+        'Frequent',
+        'Unavoidable',
+    )
+    BLIGHT_TYPES = (
+        'Spots',
+        'Gashes',
+        'Mountains',
+        'Valleys',
+        'Floating Structures',
+        'Giant Oases',
+        'Satellites',
+        'Caves',
+    )
 
+    def __init__(self):
+        # Two random planes: one primary, and one secondary
+        planes = random.sample(self.PLANES, 2)
 
-def gen_gen_population():
-    # TODO: make different percentages for population sizes
-    if random.randint(1, 10) > 3:
-        return f'{"Population:":<20}None'
+        self.plane = planes[0]
 
-    population_sizes = [
-        'Small',
-        'Medium',
-        'Large',
-        'Huge',
-        'Giant',
-    ]
-    choice = random.choice(population_sizes)
-    return f'{"Population:":<20}{choice}'
+        # Chance that the plane contains traces of another one
+        if random.randint(1, 5) > 4:
+            self.blight = planes[1]
+            self.blight_type = random.choice(self.BLIGHT_TYPES)
+            self.blight_frequency = random.choice(self.BLIGHT_FREQUENCIES)
+        else:
+            self.blight = None
+            self.blight_type = None
+            self.blight_frequency = None
+
+    def __iter__(self):
+        if self.blight is not None:
+            yield 'Mainly', self.plane
+            yield 'Deviation', f'{self.blight_frequency} {self.blight_type} of {self.blight}'
+        else:
+            yield 'Plane', self.plane
+
+
+class PlanetEconomy(PlanetAttribute):
+    """Planet economy generator"""
+
+    ECONOMIES = (
+        'Raw materials',
+        'Merchant',
+        'Black market',
+        'None',
+    )
+
+    def __init__(self):
+        self.economy = random.choice(self.ECONOMIES)
+
+    def __iter__(self):
+        yield 'Economy', self.economy
 
 
-def gen_conflict():
-    if random.randint(1, 10) > 5:
-        return f'{"Conflict:":<20}None'
+class PlanetConflict(PlanetAttribute):
+    """Planet conflict generator"""
 
-    conflict_types = [
+    CONFLICT_TYPES = (
         'War',
         'Political land boundaries',
         'Citizen land boundaries',
         'Theft',
         'Crime - gangs',
         'Crime - slavery',
         'Crime - domestic abuse',
-        'Politial power struggle',
-    ]
-    choice = random.choice(conflict_types)
-    return f'{"Conflict:":<20}{choice}'
+        'Political power struggle',
+    )
 
+    def __init__(self):
+        self.conflict = random.choice(self.CONFLICT_TYPES)
 
-def gen_raw_materials():
-    if random.randint(1, 10) > 8:
-        return f'{"Raw materials:":<20}None'
+    def __iter__(self):
+        yield 'Conflict', self.conflict
 
-    raw_materials = [
+
+class PlanetPopulation(PlanetAttribute):
+    """Planet population generator"""
+
+    # TODO: make different percentages for population sizes
+    POPULATION_SIZES = (
+        'Small',
+        'Medium',
+        'Large',
+        'Huge',
+        'Giant',
+    )
+
+    def __init__(self):
```

Comment:
interesting way to do the dependent attributes -> instantiate all dependent attrs when instantiating the thing that opens the door for those.

I think this might be the main reason for a class-based attribute structure. will think on it